parlai/crowdsourcing/utils/analysis.py

```diff
@@ -79,3 +82,65 @@ def __init__(self, opt: Dict[str, Any]):
             raise ValueError(
                 'There must be a "none_all_good" category in self.problem_buckets!'
             )
+
+    @abstractmethod
+    def compile_results(self) -> pd.DataFrame:
+        """
+        Method for returning the final results dataframe.
+        """
+
+
+class AbstractDataBrowserResultsCompiler(AbstractResultsCompiler):
+    """
+    Results compiler subclass to provide utility code for crowdsourcing tasks.
```

Comment:
nit: would be good to say something specifically about the fact that this is a DataBrowser-based subclass in the first line of the docstring, because otherwise this first line on its own doesn't say anything about what the subclass is